pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -191,13 +191,15 @@ func (dr *DashboardServiceImpl) BuildSaveDashboardCommand(ctx context.Context, d
 
 	// Validate folder
 	if dash.FolderUID != "" {
-		folder, err := dr.folderStore.GetFolderByUID(ctx, dash.OrgID, dash.FolderUID)
-		if err != nil {
-			return nil, err
+		if !dr.features.IsEnabledGlobally(featuremgmt.FlagKubernetesFoldersServiceV2) {
```

Comment:
Looks like we're missing the logic for when the flag is enabled here.

Reply:
For now, we are going to skip this part.

In this concrete case, the `folder` is being retrieved just to get the `folder.id` on line 201

we are going to address the handling of Folder.ID in a follow up PR, after merging this one.

Reply:
If we're addressing this in a follow-up PR, my preference would be to not change it yet, as this would break the dashboard logic when the feature toggle is enabled.